createDataset.py

```diff
@@ -25,162 +151,74 @@ def getWhatsAppData():
-	                otherPersonsMessage = cleanMessage(otherPersonsMessage)
-	                myMessage = cleanMessage(myMessage)
-	                responseDictionary[otherPersonsMessage] = myMessage
-	                break
-	            otherPersonsMessage = justMessage[colon+2:] + otherPersonsMessage
-	        myMessage, otherPersonsMessage, currentSpeaker = "","",""    
-	return responseDictionary
-
-def getLinkedInData():
-	df = pd.read_csv('Inbox.csv')
-	dateTimeConverter = lambda x: datetime.strptime(x,'%B %d, %Y, %I:%M %p')
-	responseDictionary = dict()
-	peopleContacted = df['From'].unique().tolist()
-	for person in peopleContacted:
-	    receivedMessages = df[df['From'] == person]
-	    sentMessages = df[df['To'] == person]
-	    if (len(sentMessages) == 0 or len(receivedMessages) == 0):
-	        # There was no actual conversation
-	        continue
-	    combined = pd.concat([sentMessages, receivedMessages])
-	    combined['Date'] = combined['Date'].apply(dateTimeConverter)
-	    combined = combined.sort(['Date'])
-	    otherPersonsMessage, myMessage = "",""
-	    firstMessage = True
-	    for index, row in combined.iterrows():
-	        if (row['From'] != personName):
-	            if myMessage and otherPersonsMessage:
-	                otherPersonsMessage = cleanMessage(otherPersonsMessage)
-	                myMessage = cleanMessage(myMessage)
-	                responseDictionary[otherPersonsMessage.rstrip()] = myMessage.rstrip()
-	                otherPersonsMessage, myMessage = "",""
-	            otherPersonsMessage = otherPersonsMessage + row['Content'] + " "
-	        else:
-	            if (firstMessage):
-	                firstMessage = False
-	                # Don't include if I am the person initiating the convo
-	                continue
-	            myMessage = myMessage + str(row['Content']) + " "
-	return responseDictionary
+def getDiscordData():
+    personName = raw_input('Enter your full Discord name: ')
+    # Putting all the file names in a list
+    allFiles = []
+    # Edit these file and directory names if you have them saved somewhere else
+    for filename in os.listdir('DiscordChatLogs'):
+        if filename.endswith(".txt"): 
+            allFiles.append('DiscordChatLogs/' + filename)
+    responseDictionary = dict() # The key is the other person's message, and the value is my response
+    # Going through each file, and recording everyone's messages to me, and my
+                                # responses
+    for currentFile in allFiles:
+        with open(currentFile, 'r') as openedFile:
+            allLines = openedFile.readlines()
+        data = ''.join(allLines)
+        otherPersonsMessage, myMessage = "",""
+        response_sets = re.findall(r'\[.+\] (?!' + re.escape(personName) + r').+\n(.+)\n{2}(?:\[.+\] ' + re.escape(personName) + r'\n(.+)\n{2})', data)
+        for response_set in response_sets:
+            otherPersonsMessage = response_set[0]
+            myMessage = response_set[1]
```

Comment:
I don't have as much experience with `re` but just wanted to make sure that `response_set` will always have at least 2 elements in it? If not, probably want some sort of check or graceful error handling.

Reply:
yes if the match is *found*, it must have 2 elements. I've tested this one before, should be fine